invenio_github/models.py

```diff
@@ -220,29 +226,32 @@ def disable(cls, user_id, github_id, name):
         :param repo_id: GitHub id of the repository.
         :param name: Fully qualified name of the repository.
         """
-        repo = cls.get(user_id, github_id=github_id, name=name)
-        if repo:
-            repo.enabled = False
-            repo.user_id = None
+        try:
+            repo = cls.get(user_id, github_id=github_id, name=name)
             repo.hook = None
-            db.session.add(repo)
+            repo.user_id = None
+        except NoResultFound:
```

Comment:
But in the context of a call to the `Repository.disable` API method, you just want to end up with not having a specific repository "enabled" in the database. If it doesn't exist then you're fine, it still counts as "disabled".